[Tooling] Update GitHub Actions

iangmaia opened the pull request:
Reference: p7H4VZ-59z-p2

Simplenote Android doesn't use one of the GitHub Actions about to have its older versions deprecated as `actions/upload-artifact`, `actions/download-artifact` or `actions/cache`, but I used this opportunity to upgrade the Action `gradle/actions/wrapper-validation` and to move the Buildkite Gradle Wrapper Validation to the Linter queue.

iangmaia committed:
Update Gradle Wrapper Validation GitHub Action

iangmaia committed:
Run Buildkite Gradle Wrapper Validation on the Linter queue

spencertransier commented:
@iangmaia Approved to unblock, but is there a reason this change would affect the linting errors that Danger picked up?

iangmaia committed:
Fix Rubocop violations

iangmaia commented:
> @iangmaia Approved to unblock, but is there a reason this change would affect the linting errors that Danger picked up?

It's not related to these changes but they might have been dormant for some reason -- my guess is that the Linter didn't pick them up in a previous PR due to a bug (now fixed) that it would use a different Rubocop version than the one configured in the project Bundle.
I confirmed locally that Rubocop reports the same issues, so I fixed them and pushed on a3a00bd6b1abb68237a6164e6474def159ec7bce.
